src/components/buttons/icon-button.tsx

```diff
@@ -0,0 +1,35 @@
+import { lighten } from 'polished';
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
+  border: none;
+  border-radius: 30px;
+  cursor: pointer;
+
+  height: 24px;
+  width: 24px;
+
+  background: linear-gradient(275.41deg, #543fd7 0%, #2756fd 100%);
+
+  color: #ffffff;
+
+  padding: 0;
+
+  &:hover {
+    background: #3e61ee;
+  }
+
+  &:active {
+    background: ${lighten(0.1, '#3e61ee')};
+  }
+`;
+
+interface CopyProps {
+  onClick: React.MouseEventHandler;
+  children: React.ReactNode;
+}
+
+export const IconButton = ({ onClick, children }: CopyProps) => {
+  return <StyledButton onClick={onClick}>{children}</StyledButton>;
```

Comment:
You only expect to receive an icon through children. You never expect more. Why would you use the children? There is no rule to use children so I would omit it. Instead I would try a similar technique as with the illustrations:

1. Hard code the icon here and only pass in the icon name
2. The prop for the icon can be typed similar to how the accordion types the illlustration type. Great for auto completion and icon safety.

See the accordion:

https://github.com/satellytes/satellytes.com/blob/5cc2990d9777d0791d165ae8ca12570fcf59ab0d/src/components/accordion/accordion.tsx#L36

Reply:
Nice idea, way cleaner, I have done it like that 👍